### PR TITLE
Vnet integration for ADF, KV & Databricks

### LINKED ISSUE #33: VNet integration for Databricks and ADF
As a Security Officer,
I need to make sure that the Databricks and ADF deployments are secure
So that patient's data is safe 

Acceptance criteria:
- VNet integration for Databricks and ADF

Note on implementation: 
There was similar work done recently in AzureTRE repo: https://github.com/microsoft/AzureTRE/pull/1857/files

### PR DESCRIPTION
**Pull request by jjgriff93**
Resolves #33 

- Managed virtual network for Databricks with secure cluster connectivity
- Managed virtual network for ADF
- Vnet integration for KeyVault
- ADF Linked service configuration for KeyVault
- Databricks secrets scope integration with KeyVault
- Remove unnecessary Databricks provider from core and serve modules

### CODE REVIEW

**Review by tanya-borisova — APPROVED** [review]

File: infrastructure/transform/main.tf [hunk 1]
```diff
@@ -47,31 +57,79 @@ resource "databricks_cluster" "fixed_single_node" {
...
 
-resource "azurerm_role_assignment" "adf_can_create_clusters" {
-  scope                = azurerm_databricks_workspace.databricks.id
-  role_definition_name = "Contributor"
-  principal_id         = azurerm_data_factory.adf.identity[0].principal_id
+resource "azurerm_role_assignment" "databricks_kv_backed_secret_scopes" {
+  scope                = var.core_kv_id
+  role_definition_name = "Key Vault Secrets User"
+
+  // This is fixed for the AzureDatabricks provider.
```
[the remark this answers is not in the document]
Comment (jjgriff93) [reply infrastructure/transform/main.tf:69]: Good spot, thanks!

**Comment by t-young31** [comment]
😢 

```
azurerm_role_assignment.deployer_can_administrate_kv: Creating...
╷
│ Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'x' with object id 'x' does not have authorization to perfxxorm action 'Microsoft.Authorization/roleAssignments/write' over scope '/subscriptions/xxx/resourceGroups/rg-xxx-dev/providers/Microsoft.KeyVault/vaults/kv-xxx/providers/Microsoft.Authorization/roleAssignments/xxx' or the scope is invalid. If access was recently granted, please refresh your credentials."
```

**Comment by jjgriff93** [comment]
> 😢
> 
> ```
> azurerm_role_assignment.deployer_can_administrate_kv: Creating...
> ╷
> │ Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'x' with object id 'x' does not have authorization to perfxxorm action 'Microsoft.Authorization/roleAssignments/write' over scope '/subscriptions/xxx/resourceGroups/rg-xxx-dev/providers/Microsoft.KeyVault/vaults/kv-xxx/providers/Microsoft.Authorization/roleAssignments/xxx' or the scope is invalid. If access was recently granted, please refresh your credentials."
> ```

Well that's a new one! Which subscription are you deploying to and can you heck what roles you're assigned?

**Comment by jjgriff93** [comment]
As discussed in chat you'll need to be owner to assign role permissions

**Review by t-young31 — APPROVED** [review]
LGTM 🚀

**Pull request merged**